casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -151,7 +151,9 @@ object Validate {
       for {
         _ <- Log[F].warn(ignore(b, s"block post state is missing."))
       } yield false
-    } else if (b.body.get.deploys.flatMap(_.log).exists(_.eventInstance == EventInstance.Empty)) {
+    } else if (b.body.get.deploys
+                 .flatMap(_.deployLog)
+                 .exists(_.eventInstance == EventInstance.Empty)) {
```

Comment:
same should be done for payment log

Reply:
Hmm. Right now, all of the payment logs are empty because no payment is occurring. Also, in `RuntimeManager`, it looks like the `InternalProcessedDeploy` corresponding to the `DeployData` is added to the output sequence even if an error is detected. What if a deploy doesn't have enough REV to pay for the deploy? Then, no deploy is processed, so no payment is made, so the payment log is empty. Is that something worth logging?

Reply:
Actually, doesn't that mean that we should allow empty event instances for the regular deploy log as well? With payments, it's no longer the case that deploys always result in a non-empty deploy log.

Reply:
Come to think of it, was never the case that deploys result in non-empty deploy logs. Consider the deployment `1 + 1`. Since it doesn't affect the store, there are no criteria against which we can validate it. Still, it's a valid deploy - albeit one that does not have a non-empty deploy log.

Reply:
You're confusing the log case (en empty list - which is fine) with a non-empty list containing the nonsensical protobuf garbage that FooInstance.Empty is. If this ever happens, this simply means we're talking to an incompatible node.